qp_klp/process_metagenomics_job.py

```diff
@@ -245,17 +246,22 @@ def process_metagenomics(sample_sheet_path, lane_number, qclient,
 
     if not skip_exec:
         gpf_job.run(callback=status_line.update_job_step)
-
-    results = map_sample_names_to_tube_ids(sn_tid_map_by_project,
-                                           join(pipeline.output_path,
-                                                'GenPrepFileJob',
-                                                'PrepFiles'))
-
-    for project in results:
-        for prep_file in results[project]:
-            df = results[project][prep_file]
-            # write modified results back out to file
-            df.to_csv(prep_file, index=False, sep="\t")
+        # if seq_pro is run, prep_file_paths will be populated by run().
+        prep_file_paths = gpf_job.prep_file_paths
+        # concatenate the lists of paths across all study_ids into a single
+        # list.
+        pfp_list = list(chain.from_iterable(prep_file_paths.values()))
+    else:
+        # under testing conditions, gpf_job.prep_file_paths will not be
+        # populated. Generate pfp_list from walking the expected location.
+        pfp_list = []
+        for root, dirs, files in walk(join(pipeline.output_path,
+                                           'GenPrepFileJob', 'PrepFiles')):
+            for prep_file in files:
```

Comment:
This looks like it's looking for files, is that right? I thought the plan was to not do that, no?

Reply:
The issue is that skip_exec is True during testing and GenPrepFileJob.run() and thus seqpro will not execute. If seqpro doesn't execute there won't be a list of known prep-file outputs. Hence, searching for prep-files in the designated location is the only way to have a generalized solution for the pipeline.